### ISSUE TITLE
Electron 5.0 renderer crashes when launching signed OSX release

### ISSUE THREAD

**Issue by thomasnordquist** [electron/electron#18066]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v5.0.0
* **Operating System:**
  * Mac OSX 10.14.1
* **Last Known Working Electron version:**:
  * 4.1.13

### Expected Behavior
The renderer starts.

### Actual Behavior
The renderer crashes on start.

```bash
/Volumes/MQTT\ Explorer\ 0.2.5/MQTT\ Explorer.app/Contents/MacOS/MQTT\ Explorer 
subscribing app/update/check
[19:02:08.080] [info] App starting...
subscribing connection/add/mqtt
subscribing connection/remove
subscribing storage/store
subscribing storage/load
subscribing storage/clear
icon path /Volumes/MQTT Explorer 0.2.5/MQTT Explorer.app/Contents/Resources/app.asar/dist/src/icon.png
[24951:0430/190211.166344:FATAL:gpu_data_manager_impl_private.cc(892)] The display compositor is frequently crashing. Goodbye.
[1]    24951 trace trap
```

### Notes
- During development the issue cannot be observed. (Electron installed via npm)
- After codesigning and packaging the BrowserWindow cannot be opened.
- Codesigning works
- Electron builder should download the same Electron executable. (should!)
- launching with `--no-sandbox` solves the issue.
- Related to https://github.com/electron/electron/issues/16631
- Looked in the electron-builder issue tracker to if it is a known issue with electron-builder

### To Reproduce
May be hard to reproduce since the app needs to be packaged. (signed ?!)
https://github.com/thomasnordquist/MQTT-Explorer

On mac:
```
git checkout 9e71c3132eda90f855746059413eae8c67cc8366
yarn
yarn prepare-release
yarn package mac
./build/clean/build/mac/MQTT\ Explorer.app/Contents/MacOS/MQTT\ Explorer
```

Workaround: https://github.com/thomasnordquist/MQTT-Explorer/commit/86544c13340573906335bc828a67d72a647c8f54

**Comment by nornagon** [comment]
This is likely due to the fact that we enable mixed-sandbox mode by default starting in Electron 5, which means the GPU process is sandboxed. I'm surprised that codesigning would cause the sandbox to fail, though! I'll look into this.

**Comment by thomasnordquist** [comment]
May not just be codesigning. There is also `electron-builder` in the mix, which downloads not the npm package but the electron github release.

**Comment by nornagon** [comment]
Hm, seems like codesigned apps on Electron 5.0.0 do launch correctly. It seems like this might be electron-builder related. Can you reproduce the issue purely with `codesign` commands?

**Comment by codebytere** [comment]
We haven't gotten a response to our questions in our comment [above](https://github.com/electron/electron/issues/18066#issuecomment-488053668). With only the information that is currently in the issue, we don't have enough information to take action. In this event, i'm going to go ahead and close this but can reopen should you follow up with more info!

**Comment by jack-arms** [comment]
I'm facing a similar renderer issue:
```
[43057:0909/152657.401905:ERROR:gl_implementation.cc(280)] Failed to load /Users/jackarms/XXX.app/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libswiftshader_libGLESv2.dylib: dlopen(/Users/jackarms/XXX.app/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libswiftshader_libGLESv2.dylib, 1): image not found
[43062:0909/152657.662180:ERROR:gl_implementation.cc(280)] Failed to load /Users/jackarms/XXX.app/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libswiftshader_libGLESv2.dylib: dlopen(/Users/jackarms/XXX.app/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libswiftshader_libGLESv2.dylib, 1): image not found
[43064:0909/152657.920735:ERROR:gl_implementation.cc(280)] Failed to load /Users/jackarms/XXX.app/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libswiftshader_libGLESv2.dylib: dlopen(/Users/jackarms/XXX.app/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libswiftshader_libGLESv2.dylib, 1): image not found
[43052:0909/152658.931688:FATAL:gpu_data_manager_impl_private.cc(892)] The display compositor is frequently crashing. Goodbye.
Trace/BPT trap: 5
```

So it seems like it's looking for `libswiftshader_libGLESv2.dylib` which was added in bundles by https://github.com/electron/electron/commit/23286fe5573fdb0e3b263d849dbc2bcc9746776f and I see that it's in the zip for the latest 6.x release. Anyone know if there's a way to fix this issue on 5.0.8 or do I have to upgrade to 6.x to have that library bundled? Running with `--no-sandbox` does not produce this issue, so I can't tell if in that case if:
- it doesn't try to load the library
- it tries to load and fails silently
- it successfully loads it where it comes from if it's not bundled.

**Comment by ahmadwaliesipick** [comment]
@jack-arms  are you able to resolved issue? i am still facing same error.
